filecoin/retriever.go

```diff
@@ -68,14 +66,16 @@ type Retriever struct {
 	// Assumed immutable during operation
 	config RetrieverConfig
 
-	endpoint  Endpoint
-	filClient *filclient.FilClient
+	endpoint     Endpoint
+	filClient    *filclient.FilClient
+	eventManager *eventManager
 
-	runningRetrievals        map[cid.Cid]bool
-	activeRetrievalsPerMiner map[peer.ID]uint
-	runningRetrievalsLk      sync.Mutex
+	activeRetrievals *activeRetrievalsManager
 
 	minerMonitor *minerMonitor
+
+	runningRetrievals   map[cid.Cid]bool
+	runningRetrievalsLk sync.Mutex
```

Comment:
may want to look at an RWMutex for this

Reply:
oh maybe never mind this is something different I think.

Reply:
a debugging piece I'm thinking of removing

Reply:
removed